components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2768,18 +2780,32 @@ void RewardsServiceImpl::OnFetchBalance(
     const ledger::type::Result result,
     ledger::type::BalancePtr balance) {
   if (balance) {
+    PrefService* pref_service = profile_->GetPrefs();
+
     if (balance->total > 0) {
-      profile_->GetPrefs()->SetBoolean(prefs::kUserHasFunded, true);
+      pref_service->SetBoolean(prefs::kUserHasFunded, true);
     }
 
-    // Record stats.
-    double balance_minus_grant = CalcWalletBalanceForP3A(
-        balance->wallets,
-        balance->user_funds);
-    RecordWalletBalanceP3A(
-        true,
-        true,
-        static_cast<size_t>(balance_minus_grant));
+    // Record wallet state stats
+    if (IsRewardsEnabled()) {
+      if (pref_service->GetBoolean(prefs::kUserHasClaimedGrant)) {
+        p3a::RecordWalletState({.wallet_created = true,
+                                .rewards_enabled = true,
+                                .grants_claimed = true,
+                                .funds_added = balance->user_funds > 0});
+      } else {
+        p3a::RecordWalletState({.wallet_created = true,
+                                .rewards_enabled = true,
+                                .grants_claimed = false,
```

Comment:
Nit: it looks like these two branches can be combined now, if you want, with:

```cpp
.grants_claimed = pref_service->GetBoolean(prefs::kUserHasClaimedGrant),
```

Reply:
Ah yeah, will do.